fortuna/prob_model/posterior/sgmcmc/sgmcmc_posterior_mixin.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from copy import deepcopy
+from typing import Optional, Tuple, Type
+import pathlib
+
+from jax._src.prng import PRNGKeyArray
+from jax import pure_callback, random
+
+from fortuna.prob_model.posterior.state import PosteriorState
+from fortuna.prob_model.fit_config.base import FitConfig
+from fortuna.prob_model.joint.state import JointState
+from fortuna.prob_model.posterior.map.map_state import MAPState
+from fortuna.prob_model.posterior.posterior_multi_state_repository import \
+    PosteriorMultiStateRepository
+from fortuna.typing import Path
+
+logger = logging.getLogger(__name__)
+
+
+class SGMCMCPosteriorMixin:
+    """Mixin class that handles sampling from SGMCMC posterior approximators."""
+    def sample(
+        self,
+        rng: Optional[PRNGKeyArray] = None,
+        **kwargs,
+    ) -> JointState:
+        """
+        Sample from the posterior distribution.
+
+        Parameters
+        ----------
+        rng : Optional[PRNGKeyArray]
+            A random number generator. If not passed, this will be taken from the attributes of this class.
+
+        Returns
+        -------
+        JointState
+            A sample from the posterior distribution.
+        """
+        if rng is None:
+            rng = self.rng.get()
+        state = pure_callback(
+            lambda j: self.state.get(i=j),
+            self.state.get(i=0),
+            random.choice(rng, self.posterior_approximator.n_samples),
+        )
+        return JointState(
```

Comment:
Why `JointState` and not `SGMCMCState`? Just trying to understand.

Reply:
It's a mixin for `Posterior` subclasses, so following the contract here: https://github.com/awslabs/fortuna/blob/main/fortuna/prob_model/posterior/base.py#L125 Maybe it would be clearer to refactor it into a base class instead of mixin?